src/App.tsx

```diff
@@ -34,62 +34,88 @@ export function getReorderedGoods(
   // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  if (sortType === 1) {
+    visibleGoods.sort((a, b) => a.localeCompare(b));
+  } else if (sortType === 2) {
+    visibleGoods.sort((a, b) => a.length - b.length);
+  }
 
-  return visibleGoods;
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
 }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
 
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
+  render() {
+    const goods = getReorderedGoods(
+      goodsFromServer,
+      this.state,
+    );
 
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={`button is-info${this.state.sortType !== 1 ? ' is-light' : ''}`}
+            onClick={() => this.setState({ sortType: 1 })}
+          >
+            Sort alphabetically
+          </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+          <button
+            type="button"
+            className={`button is-success${this.state.sortType !== 2 ? ' is-light' : ''}`}
+            onClick={() => this.setState({ sortType: 2 })}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={`button is-warning${!this.state.isReversed ? ' is-light' : ''}`}
+            onClick={() => this.setState(prev => ({
+              isReversed: !prev.isReversed,
+            }))}
+          >
+            Reverse
+          </button>
+
+          {this.state.sortType !== 0 || this.state.isReversed
+            ? (
+              <button
+                type="button"
+                className="button is-danger is-light"
+                onClick={() => this.setState({
+                  isReversed: false,
+                  sortType: 0,
+                })}
+              >
+                Reset
+              </button>
+            )
+            : <></>}
```

Comment:
and overall it's even easier - just conditionally render the whole button - without ternary, but with `&&`
```suggestion
          {this.state.sortType !== 0 || this.state.isReversed
            && (
              <button
                type="button"
                className="button is-danger is-light"
                onClick={() => this.setState({
                  isReversed: false,
                  sortType: 0,
                })}
              >
                Reset
              </button>
            )}
```

Reply:
Sometimes I look back at my code, and ask myself why..

Looks like because I take breaks very rarely